docs/RationaleHWArith.md

```diff
@@ -0,0 +1,71 @@
+# HW Arith Dialect Rationale
+
+This document describes the various design points of the HWArith dialect, a dialect that is used to represent bitwidth aware arithmetic. This follows in the spirit of other
+[MLIR Rationale docs](https://mlir.llvm.org/docs/Rationale/).
+
+- [HW Arith Dialect Rationale](#hw-arith-dialect-rationale)
+  - [Introduction](#introduction)
+    - [Q&A](#qa)
+  - [Bit Width Rules](#bit-width-rules)
+  - [Lowering](#lowering)
+
+## Introduction
+
+The `hwarith` dialect provides a collection of operations that define typical integer arithmetic operations which are bitwidth aware.
+This bitwidth awareness is expressed through return type inference rules that, based on the types of an operation and its input operands, infers the type of the result operand.
+
+In general, the bit width inference rules are designed to model SystemVerilog semantics. However, unlike SystemVerilog, the aim of this dialect is to provide a strongly-typed hardware arithmetic library with the width inference rules for every operator explained in a clear and understandable way to avoid ambiguity.
```

Comment:
Nit: I generally limit lines to 80 cols just like code.

Reply:
I'm usually trying to push for long-lines-with-word-wrap since I find it very painful to edit 80-cols formatted text (and, that I never work with textual documents in terminals). But, that is a battle I'll have to fight elsewhere - I'll change it to 80 cols just to be consistent with the remainder of the rationales.

Reply:
Yeah, I got used to that pain in grad school. Makes reviewing changes painful as well. But long lines make my eyes hurt.